### PR TITLE
Update dependencies and fix deprecations

### PR DESCRIPTION
**Pull request by devmotion**
This PR updates the dependencies of Bijectors, adds package compatibilities (I took lower bounds from Turing), and fixes some deprecations.

### CODE REVIEW

**Comment by devmotion** [comment]
The updates fix the current errors due to Pkg dependencies on Julia nightly (see, e.g., https://ci.appveyor.com/project/TuringLang/bijectors-jl/builds/29787947/job/tegs9jryfyla563a) but expose a type inference problem on Julia nightly.

File: src/bijectors/stacked.jl [hunk 1]
```diff
@@ -80,12 +80,15 @@ function (sb::Stacked{<:AbstractArray, N})(x::AbstractVector{<:Real}) where {N}
     return y
 end
 
-# HACK: `reshape` to get around the fact that `hcat` isn't type-stable
-(sb::Stacked)(x::AbstractMatrix{<:Real}) = reshape(foldl(hcat, [sb(x[:, i]) for i = 1:size(x, 2)]), size(x)...)
+@views function (sb::Stacked)(x::AbstractMatrix{<:Real})
+    init = reshape(sb(x[:, 1]), :, 1)
+    return mapreduce(i -> sb(x[:, i]), hcat, 2:size(x, 2); init = init)
```
Comment (mohamed82008) [inline src/bijectors/stacked.jl]: No need to define `init` separately. This is equivalent to using `1:size(x, 2)` and dropping the `init` kwarg.

**Comment by mohamed82008** [comment]
Oh and thanks for the PR btw :)

File: src/bijectors/stacked.jl (hunk 1 above)
Comment (mohamed82008) [reply src/bijectors/stacked.jl]: Hmm actually I double checked and your implementation here is type stable but the compiler is struggling to infer my suggestion.

Comment (devmotion) [reply src/bijectors/stacked.jl]: Yes, this change is only for fixing the type inference problem.

**Comment by devmotion** [comment]
I'm in favour of removing the Manifest file and not depending on ForwardDiff and Tracker as well :+1: Actually I was wondering about the Manifest file and the Requires approach, but since I didn't know what's the general opinion about it I just stuck to the current approach (with Manifest file and ForwardDiff and Tracker as hard dependencies - even though it might seem otherwise due to the dependency on Requires and the optional block in `Bijectors.jl` ForwardDiff and Tracker are already dependencies and loaded in `interface.jl`).

**Comment by cpfiffer** [comment]
I've been removing the Manifest everywhere else -- I think it's bad form to leave it in the repo.

File: src/bijectors/scale.jl [hunk 2]
```diff
@@ -24,62 +24,4 @@ _logabsdetjac_scale(a::Real, x::AbstractMatrix, ::Val{1}) = fill(log(abs(a)) * s
 _logabsdetjac_scale(a::AbstractVector, x::AbstractVector, ::Val{1}) = sum(log.(abs.(a)))
 _logabsdetjac_scale(a::AbstractVector, x::AbstractMatrix, ::Val{1}) = fill(sum(log.(abs.(a))), size(x, 2))
 _logabsdetjac_scale(a::AbstractMatrix, x::AbstractVector, ::Val{1}) = log(abs(det(a)))
-_logabsdetjac_scale(a::AbstractMatrix, x::AbstractMatrix{T}, ::Val{1}) where {T} = log(abs(det(a))) * ones(T, size(x, 2))
```
Comment (torfjelde) [inline src/bijectors/scale.jl]: Is there a reason why this line was removed?

Comment (devmotion) [reply src/bijectors/scale.jl]: It's not actually removed, somehow I just removed the newline character at the end of the line.

Comment (torfjelde) [reply src/bijectors/scale.jl]: Ah, np then:)

**Comment by mohamed82008** [comment]
I like the `compat` idea.

**Comment by devmotion** [comment]
I like this idea as well, so I moved the optional functionality to a `compat` directory.

**Pull request merged**